runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java

```diff
@@ -138,11 +150,20 @@ static StreamExecutionEnvironment createStreamExecutionEnvironment(
     } else if (masterUrl.matches(".*:\\d*")) {
       List<String> parts = Splitter.on(':').splitToList(masterUrl);
       flinkConfig.setInteger(RestOptions.PORT, Integer.parseInt(parts.get(1)));
+
+      SavepointRestoreSettings savepointRestoreSettings = SavepointRestoreSettings.none();
+      if (options.getSavepointPath() != null) {
+        savepointRestoreSettings =
+            SavepointRestoreSettings.forPath(
+                options.getSavepointPath(), options.getAllowNonRestoredState());
+      }
```

Comment:
Maybe it's just me but I prefer this pattern:

```java
      final SavepointRestoreSettings savepointRestoreSettings;
      if (options.getSavepointPath() != null) {
        savepointRestoreSettings =
            SavepointRestoreSettings.forPath(
                options.getSavepointPath(), options.getAllowNonRestoredState());
      } else {
        savepointRestoreSettings = SavepointRestoreSettings.none();
      }
```

It's slightly more verbose but also more robust.

Reply:
It's probably more a matter of taste, I made the change.